index.bs

```diff
@@ -259,9 +259,21 @@ A <dfn>lock request</dfn> represents a pending request for a [=lock-concept|lock
 
 A [=lock request=] is a [=/struct=] with [=struct/items=] <dfn>agent</dfn>, <dfn>clientId</dfn>, <dfn>origin</dfn>, <dfn>name</dfn>, <dfn>mode</dfn>, and <dfn>promise</dfn>.
 
-A <dfn>lock request queue</dfn> is a [=/queue=] of [=lock requests=].
+</div>
+
+A <dfn>lock request queue</dfn> is a [=/queue=] of [=/lock requests=].
+
+Each origin has a <dfn>lock request queue map</dfn>, which is a [=map=] of [=resource names=] to [=/lock request queues=].
+
+<div algorithm>
+
+To <dfn>get the lock request queue</dfn> from [=/lock request queue map=] |queueMap| from [=/resource name=] |name|, run these steps:
+
+1. If |queueMap|[|name|] does not [=map/exist=], [=map/set=] |queueMap|[|name|] to a new empty [=/lock request queue=].
+1. Return |queueMap|[|name|].
```

Comment:
Cool, I guess this can ultimately be upstreamed to infra spec as something like C++ [`map::emplace()`](https://www.cplusplus.com/reference/map/map/emplace/).